src/extension.ts

```diff
@@ -1023,8 +1026,17 @@ export async function renameDataSet(node: IZoweDatasetTreeNode, datasetProvider:
             datasetProvider.renameFavorite(node, afterDataSetName);
             node.label = temp;
         }
-        datasetProvider.refreshElement(node.getParent());
+
+        datasetProvider.refreshElement(node);
         datasetProvider.updateFavorites();
+
+        if (fs.existsSync(beforeFullPath)) {
```

Comment:
In general `sync` methods are not a good idea, because they block the event loop. Perhaps someone from the team can comment on whether that is or is not the case here?

Reply:
🤔 Interesting comment.
While I agree that synchronous event block the event loop, it's also true that we logically need to `await` for the code to execute.

I don't see it as a bad practice to use `async` methods whenever possible, even though the logical structure seems the same.
In the long run, we might get a slight performance increase thanks to the JS engine and someone could make the case that it's theoretically more secure because of blocked threads due to malicious input.

As a general rule, (personally) I don't request changes if the functionality does not what's supposed to, or conflicts with other existing design paradigm.

Hope this helps 😄

Reply:
Looking at the code around and thinking a bit more about this, I believe we should stick to a single paradigm - unless there is a reason to do things otherwise. All of our code uses async/await so I believe it would be better to use it here as well.
```suggestion
import {promisify} from 'util';
const exists = util.promisify(fs.exists);
        if (await exists(beforeFullPath)) {
```
and similarly for `fs.unlink` bellow

Reply:
How about we tackle this as a separate (refactoring) ticket.

That way we can thoroughly go through the code and focus on the places where we don't need to be blocking the event loop. :smile:

Will open another issue for this.

And maybe we can start using the wiki? (Credits to @Colin-Stone for suggesting 👍 )